wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettingsMetamask.vue

```diff
@@ -151,7 +144,16 @@ export default {
           this.$root.$emit('wallet-settings-provider-changed', 'INTERNAL_WALLET');
         })
         .catch(() => this.$root.$emit('wallet-settings-provider-changing', window.walletSettings.wallet.provider))
-        .finally(() => this.savingMetamaskAddress = false);
+        .finally(() => {
```

Comment:
In this case redirection will be done regardeless the operation is success or failure. Is it expected ?

Reply:
Only on success fixed!